server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/executeBash.ts

```diff
@@ -536,6 +536,19 @@ export class ExecuteBash {
         })
     }
 
+    private static decodeWinUtf(raw: string): string {
+        const buffer = Buffer.from(raw, 'binary')
+
+        let utf16 = true
+        for (let i = 1, n = Math.min(buffer.length, 32); i < n; i += 2) {
```

Comment:
I don't know how to detect if a string is utf8 or utf16, but is https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/isWellFormed valid for this purpose?

Reply:
I do not believe `isWellFormed()` will work here. It only validates UTF-16 strings already in JS memory, but our issue is earlier: determining if incoming bytes are UTF-16LE or UTF-8. By the time Node decodes these bytes into a string, they'll be "well-formed" either way.

Reply:
If `/u` forces UTF-16LE, under what circumstances would we have to fall back onto UTF-8?

Reply:
It forces it for built-in commands, but if the user is running an external program, it's free to use whatever encoding it wants